gpii/node_modules/flowManager/src/UserLogonHandlers.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * GPII User Logon State Change
+ *
+ * Copyright 2012 OCAD University
+ * Copyright 2015, 2017 Raising the Floor - International
+ * Copyright 2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.flowManager.baseHandler", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        errorResponse: {
+            funcName: "gpii.flowManager.baseHandler.errorResponse",
+            args: ["{request}", "{lifecycleManager}.userErrors", "{arguments}.0", "{arguments}.1"]
+                                                                 // message, statusCode
+        }
+    }
+});
+
+gpii.flowManager.baseHandler.errorResponse = function (that, userErrors, message, statusCode) {
+    userErrors.events.userError.fire({
+        isError: true,
+        messageKey: "KeyInFail",
+        originalError: message
+    });
+
+    that.events.onError.fire({
+        statusCode: statusCode || 500,
+        message: message || "Unknown error occurred on logon change action"
+    });
+};
+
+// Request handler for /user/:gpiiKey/proximityTriggered
+fluid.defaults("gpii.flowManager.proximityTrigger.handler", {
+    gradeNames: ["kettle.request.http", "gpii.flowManager.baseHandler"],
+    debounceTimeMs: 1500,
+    invokers: {
+        handleRequest: {
```

Comment:
Whilst this has been partially refactored, we still have the situation that it's impossible to trigger, e.g. a proximityLogon except via an HTTP request to localhost, which we need to get rid of. The entry point to this method should be refactored so it is just simply an invoker or event, and then this HTTP endpoint will forward to this.